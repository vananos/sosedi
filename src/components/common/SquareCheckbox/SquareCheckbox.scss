@import "../../../base.scss";

.checkbox {
    display: flex;
    align-items: center;
    justify-content: center;
    box-sizing: border-box;
    background-color: $light-grey;
    color: $grey;
    border: 10px solid $light-grey;
    cursor: pointer;
    text-transform: uppercase;
    font-size: 1rem;
    font-weight: bold;

    input {
        display: none;
    }

    &-checked {
        background: url("../../../assets/profile/check-solid.svg") $brown;
        color: rgb(255, 255, 255);
        border: 10px solid $light-brown;
    }
}