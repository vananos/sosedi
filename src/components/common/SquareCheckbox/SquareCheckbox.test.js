import { shallow } from "enzyme";
import React from "react";
import SquareCheckbox from "./SquareCheckbox";

describe("Checkbox", () => {
  const checkboxProps = () => ({
    name: "test-checkbox",
    value: "test?"
  });

  it("checkbox is not checked, click -> checked", () => {
    const checkboxWrapper = shallow(<SquareCheckbox {...checkboxProps()} />);
    checkboxWrapper
      .find("div")
      .first()
      .simulate("click");

    expect(checkboxWrapper).toMatchSnapshot();
  });

  it("checkbox is checked, click -> not checked", () => {
    const checkboxWrapper = shallow(
      <SquareCheckbox {...checkboxProps()} checked={true} />
    );
    
    checkboxWrapper
      .find("div")
      .first()
      .simulate("click");

    expect(checkboxWrapper).toMatchSnapshot();
  });
});
