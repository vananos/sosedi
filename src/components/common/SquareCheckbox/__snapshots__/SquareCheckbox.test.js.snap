// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox checkbox is checked, click -> not checked 1`] = `
<div
  className="checkbox "
  onClick={[Function]}
  style={
    Object {
      "height": 120,
      "width": 120,
    }
  }
>
  <input
    checked={false}
    name="test-checkbox"
    onChange={[Function]}
    type="checkbox"
    value="test?"
  />
</div>
`;

exports[`Checkbox checkbox is not checked, click -> checked 1`] = `
<div
  className="checkbox checkbox-checked"
  onClick={[Function]}
  style={
    Object {
      "height": 120,
      "width": 120,
    }
  }
>
  <input
    checked={true}
    name="test-checkbox"
    onChange={[Function]}
    type="checkbox"
    value="test?"
  />
</div>
`;
